SwiftLintPlugin execution from Script Build Phase (#5677)
realm/SwiftLint: README.md

fi
```

> If you're using the SwiftLintPlugin from SPM. Use the
> following script implementation:

Reviewer: ```suggestion
If you're using the SwiftLintPlugin in a Swift package,
you may refer to the `swiftlint` executable in the
following way:
```

```bash
SWIFT_PACKAGE_DIR="${BUILD_DIR%Build/*}SourcePackages/artifacts"

> SWIFTLINT_CMD=$SWIFT_PACKAGE_DIR/swiftlintplugins/SwiftLintBinary/SwiftLintBinary.artifactbundle/swiftlint-*/bin/swiftlint

Reviewer: ```suggestion
SWIFTLINT_CMD="$SWIFT_PACKAGE_DIR/swiftlintplugins/SwiftLintBinary/SwiftLintBinary.artifactbundle/swiftlint-*/bin/swiftlint"
```

Author: Using this command
```
SWIFTLINT_CMD=$(ls "$SWIFT_PACKAGE_DIR"/swiftlintplugins/SwiftLintBinary/SwiftLintBinary.artifactbundle/swiftlint-*/bin/swiftlint | head -n 1)
```

> if test -f $SWIFTLINT_CMD 2>&1
> then

Reviewer: ```suggestion
if test -f "$SWIFTLINT_CMD" 2>&1
then
```

That's more consistent to the other Bash scripts in the README.

>     $SWIFTLINT_CMD

Reviewer: ```suggestion
    "$SWIFTLINT_CMD"
```

Quotes around variables are a good practice in case of any spaces in file paths.

Reviewer 2: I second what @SimplyDanny mentioned. Good to wrap it with spaces.

Author: Adding quote on script lines will break the script, give it a try and you will see.

Reviewer: Yeah, then the second line needs to make sure the wildcard is properly resolved. I suggest
```bash
SWIFTLINT_CMD=$(ls "$SWIFT_PACKAGE_DIR"/swiftlintplugins/SwiftLintBinary/SwiftLintBinary.artifactbundle/swiftlint-*/bin/swiftlint | head -n 1)
```
to make sure potential spaces are quoted and there is only a single result in the command.

fi
```

If you're using the SwiftLintPlugin in a Swift package,
you may refer to the `swiftlint` executable in the
following way:

```bash
SWIFT_PACKAGE_DIR="${BUILD_DIR%Build/*}SourcePackages/artifacts"
SWIFTLINT_CMD=$(ls "$SWIFT_PACKAGE_DIR"/swiftlintplugins/SwiftLintBinary/SwiftLintBinary.artifactbundle/swiftlint-*/bin/swiftlint | head -n 1)

if test -f "$SWIFTLINT_CMD" 2>&1
then
    "$SWIFTLINT_CMD"
else
    echo "warning: `swiftlint` command not found - See https://github.com/realm/SwiftLint#installation for installation instructions."
fi
```

> > [!NOTE]
> > The `SWIFTLINT_CMD` path use the default Xcode configuration and has been tested on Xcode 15/16, in case of another configuration like
> > custom Swift package path, please change the values accordinaly.

Reviewer: ```suggestion

> [!NOTE]
> The `SWIFTLINT_CMD` path uses the default Xcode configuration and has been
> tested on Xcode 15/16. In case of another configuration (e.g. a custom
> Swift package path), please adapt the values accordingly.
```
